zeppelin-web/src/components/notebookListDataFactory/notebookList.datafactory.js

```diff
@@ -14,11 +14,11 @@
 'use strict';
 
 angular.module('zeppelinWebApp').factory('notebookListDataFactory', function() {
-  var notes = {};
-  notes.root = {
-    children : []
+  var notes = {
+    root: {children: []},
+    flatList: [],
+    setNotes: function() {}
```

Comment:
@johnnyws I think you understand my comment too literally :)
`setNotes: function() {/* the function goes here */}` should actually implement the function here; defining an empty function and later redefining it is rather useless :)